src/test/java/seedu/duke/ActivityEditTest.java

```diff
@@ -1,12 +1,111 @@
 package seedu.duke;
 
 import org.junit.jupiter.api.Test;
+import seedu.duke.command.DeleteCommand;
+import seedu.duke.command.EditExerciseCommand;
+import seedu.duke.command.EditFoodCommand;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class ActivityEditTest {
+    void createObjects(ActivityList dummyList) {
+        dummyList.addActivity(new Food("Apple", 50, false));
+        dummyList.addActivity(new Food("Banana", 100, false));
+        dummyList.addActivity(new Food("Apple pie", 55, false));
+        dummyList.addActivity(new Exercise("Juggle Apples", 100, false));
+    }
+
+    @Test
+    void editActivityFoodToExercise_success() {
+        ActivityList dummyList = new ActivityList();
+        createObjects(dummyList);
+
+        Activity activity = new Exercise("jumping",900, false);
+        dummyList.insertActivity(0, activity);
+
+        assertEquals("[E] | jumping | 900, [F] | Banana | 100, [F] | Apple pie | 55, "
+                + "[E] | Juggle Apples | 100", dummyList.toString());
+    }
+
+    @Test
+    void editActivityFoodToExercise2_success() throws NullPointerException {
+        ActivityList dummyList = new ActivityList();
+        createObjects(dummyList);
+        try {
+            EditFoodCommand food = new EditFoodCommand(0, "jumping", 900);
+            food.execute();
+        } catch (NullPointerException e) {
+            System.out.println("null!");
```

Comment:
Assertions.assertThrows() might help for exception testing
An example is given here:
https://howtodoinjava.com/junit5/expected-exception-example/